scripts/build/check_init_priorities.py

```diff
@@ -77,7 +77,12 @@ def __init__(self, name):
         for idx, level in enumerate(_DEVICE_INIT_LEVELS):
             if level in name:
                 _, priority_str = name.strip("_").split(level)
-                priority, sub_priority = priority_str.split("_")
+
+                # ARCMWDT linker generates extended section names
+                # (ex.: .rela.z_init_POST_KERNEL40_0_.__init_k_sys_work_q_init).
+                # limitation is required for reading only 2 parameters from
+                # priority string.
+                priority, sub_priority = priority_str.split("_")[:2]
```

Comment:
I think it would be much safer/readable to move to regexes instead. 

```python
import re
...
# before checking each level
expr = re.compile(rf".*{level}(\d+)_(\d+).*")
...
# check if match
m = expr.match(name)
if not m:
    continue

priority, sub_priority = m.groups()
```

Reply:
@fabiobaltieri

Reply:
@kokas-a I have a PR in flight to rework this logic to not use the object files at all, so it would not rely on the format of the object sections anymore and just needs the boundary symbols in the final elf file, which should be a more robust solution (turns out the current one also breaks with GCC with LTO enabled). It deletes this code entirely, working on some kinks around armclang support but as long as the symbols are there it should work with this as well, could you give it a try? https://github.com/zephyrproject-rtos/zephyr/pull/62459

I'm also open to the idea of only running this for GCC and clang, there's only that much we can do about chasing quirks with non generally available compilers.

Reply:
@fabiobaltieri , I have finally try your changes with WMDT compiler. They work fine!

Reply:
brilliant, thanks a log, I'll nudge that one to get it unblocked then